File: ip-address-validation-nodejs/REST/ipav_response.js
export class Error {
    constructor(data = {}) {
        this.Desc = data.Desc;
        this.Number = data.Number;
        this.Location = data.Location;
    }

    toString() {
        return `Error: Desc = ${this.Desc}, Number = ${this.Number}, Location = ${this.Location}`;
    }
}

export class IPAVResponse {
    constructor(data = {}) {
        this.IPAddress = data.IPAddress;
        this.LicenseKey = data.LicenseKey;
        this.Certainty = data.Certainty || 0;
        this.City = data.City;
        this.Region = data.Region;
        this.Country = data.Country;
        this.CountryISO3 = data.CountryISO3;
        this.CountryISO2 = data.CountryISO2;
        this.PostalCode = data.PostalCode;
        this.MetroCode = data.MetroCode;
        this.DMA = data.DMA;
        this.StateFIPS = data.StateFIPS;
        this.CountyFIPS = data.CountyFIPS;
        this.Latitude = data.Latitude;
        this.Longitude = data.Longitude;
        this.IsProxy = data.IsProxy;
        this.ProxyType = data.ProxyType;
        this.PossibleMobileDevice = data.PossibleMobileDevice;
        this.ISP = data.ISP;
        this.NetblockOwner = data.NetblockOwner;
        this.HostNames = data.HostNames;
        this.IPNoteCodes = data.IPNoteCodes;
        this.IPNotes = data.IPNotes;
        this.Debug = data.Debug;
        this.Error = data.Error ? new Error(data.Error) : null;
    }

    toString() {
        return `IPAVResponse: IPAddress = ${this.IPAddress ?? 'null'}, LicenseKey = ${this.LicenseKey ?? 'null'}, ` +
               `Certainty = ${this.Certainty}, City = ${this.City ?? 'null'}, Region = ${this.Region ?? 'null'}, ` +
               `Country = ${this.Country ?? 'null'}, CountryISO3 = ${this.CountryISO3 ?? 'null'}, ` +
               `CountryISO2 = ${this.CountryISO2 ?? 'null'}, PostalCode = ${this.PostalCode ?? 'null'}, ` +
               `MetroCode = ${this.MetroCode ?? 'null'}, DMA = ${this.DMA ?? 'null'}, ` +
               `StateFIPS = ${this.StateFIPS ?? 'null'}, CountyFIPS = ${this.CountyFIPS ?? 'null'}, ` +
               `Latitude = ${this.Latitude ?? 'null'}, Longitude = ${this.Longitude ?? 'null'}, ` +
               `IsProxy = ${this.IsProxy ?? 'null'}, ProxyType = ${this.ProxyType ?? 'null'}, ` +
               `PossibleMobileDevice = ${this.PossibleMobileDevice ?? 'null'}, ISP = ${this.ISP ?? 'null'}, ` +
               `NetblockOwner = ${this.NetblockOwner ?? 'null'}, HostNames = ${this.HostNames ?? 'null'}, ` +
               `IPNoteCodes = ${this.IPNoteCodes ?? 'null'}, IPNotes = ${this.IPNotes ?? 'null'}, ` +
               `Debug = ${this.Debug ?? 'null'}, Error = ${this.Error ? this.Error.toString() : 'null'}`;
    }
}

export default IPAVResponse;

File: ip-address-validation-python/REST/ipav_response.py
from dataclasses import dataclass
from typing import Optional

@dataclass
class Error:
    Desc: Optional[str] = None
    Number: Optional[str] = None
    Location: Optional[str] = None

    def __str__(self) -> str:
        return (f"Error: Desc={self.Desc}, Number={self.Number}, Location={self.Location}")

@dataclass
class IPAVResponse:
    IPAddress: Optional[str] = None
    LicenseKey: Optional[str] = None
    Certainty: Optional[int] = None
    City: Optional[str] = None
    Region: Optional[str] = None
    Country: Optional[str] = None
    CountryISO3: Optional[str] = None
    CountryISO2: Optional[str] = None
    PostalCode: Optional[str] = None
    MetroCode: Optional[str] = None
    DMA: Optional[str] = None
    StateFIPS: Optional[str] = None
    CountyFIPS: Optional[str] = None
    Latitude: Optional[float] = None
    Longitude: Optional[float] = None
    IsProxy: Optional[str] = None
    ProxyType: Optional[str] = None
    PossibleMobileDevice: Optional[str] = None
    ISP: Optional[str] = None
    NetblockOwner: Optional[str] = None
    HostNames: Optional[str] = None
    IPNoteCodes: Optional[str] = None
    IPNotes: Optional[str] = None
    Debug: Optional[str] = None
    Error: Optional[Error] = None

    def __str__(self) -> str:
        error = str(self.Error) if self.Error else "None"
        return (f"IPAVResponse: IPAddress={self.IPAddress}, LicenseKey={self.LicenseKey}, "
                f"Certainty={self.Certainty}, City={self.City}, Region={self.Region}, "
                f"Country={self.Country}, CountryISO3={self.CountryISO3}, CountryISO2={self.CountryISO2}, "
                f"PostalCode={self.PostalCode}, MetroCode={self.MetroCode}, DMA={self.DMA}, "
                f"StateFIPS={self.StateFIPS}, CountyFIPS={self.CountyFIPS}, Latitude={self.Latitude}, "
                f"Longitude={self.Longitude}, IsProxy={self.IsProxy}, ProxyType={self.ProxyType}, "
                f"PossibleMobileDevice={self.PossibleMobileDevice}, ISP={self.ISP}, "
                f"NetblockOwner={self.NetblockOwner}, HostNames={self.HostNames}, "
                f"IPNoteCodes={self.IPNoteCodes}, IPNotes={self.IPNotes}, Debug={self.Debug}, "
                f"Error={error}")

